### PR TITLE
Migrate/Create Tests for the Setup Handler

### PR DESCRIPTION
**Pull request by xwsxethan**
I also cleaned up some unnecessary imports and constants in user_test as I noticed them when creating setup_test.

I still have more to add to this, but going ahead and making the PR while I'm thinking about it.

[Review on Reviewable](https://reviewable.io/reviews/uproxy/ufo-management-server-flask/15)

### CODE REVIEW

**Comment by xwsxethan** [comment]
@jpevarnek PTAL as well.

File: ufo/setup_test.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    args, kwargs = mock_render_template.call_args
+    self.assertEquals('setup.html', args[0])
+    self.assertEquals(self.config, kwargs['config'])
+    self.assertEquals(FAKE_OAUTH_URL, kwargs['oauth_url'])
+
+  @patch.object(discovery, 'build')
+  @patch.object(oauth, 'getOauthFlow')
+  @patch('flask.render_template')
+  def testPostSetupIncorrectDomain(self, mock_render_template,
```
Comment (blueandgold) [inline ufo/setup_test.py]: nit: can we change the method name to be more easy to read, such as:
testPostSetupWithIncorrectDomain()

**Comment by blueandgold** [comment]
Should there also be a test for when the user is not admin as well?
 if not userResult.get('isAdmin', False)

**Comment by xwsxethan** [comment]
For a non-admin test, that's what I was talking to Jonathan about yesterday. I can probably add one now, but it will basically be mocking everything haha.

As for changing the line, I modified it slightly, but kept the default value of False in case isAdmin is not found for some reason. Should still be obvious this way.

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[ufo/setup_test.py, line 57 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8WRNWmk_liDu7O2UQ_-r1-57:-K8_Ja1JUjCRg5vI7t4G:1351270098) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/d4aaa57875e4010295f9bc71b62fda84e8ae5d30/ufo/setup_test.py#L57)):*
I would have named it this actually, but it is over the 30 character limit so I had to make do. I can make WithInvalidDomain fit, but that didn't really seem that accurate. The domain isn't invalid; it just does not match the auth code. If you have another suggestion, I'm open to it though.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-:-K8_K21At1HKZKidI9iK:745344401)*

**Comment by goldspin** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[ufo/setup_test.py, line 57 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8WRNWmk_liDu7O2UQ_-r1-57:-K8_Nr18Q1RB8Lv48jxI:-36992754) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/d4aaa57875e4010295f9bc71b62fda84e8ae5d30/ufo/setup_test.py#L57)):*
I see the problem now.  :)  Technically, the test name should be testPostSetupWithDifferentDomainCredential().  But if there's a constrain, then feel free to leave it as is.

For the admin portion, that execution path should be tested.  It's okay to add later, just put a TODO.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15)*

**Comment by xwsxethan** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[ufo/setup_test.py, line 57 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8WRNWmk_liDu7O2UQ_-r1-57:-K8_XJWYziW65b3SjCkf:-1938984845) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/d4aaa57875e4010295f9bc71b62fda84e8ae5d30/ufo/setup_test.py#L57)):*
I implemented that test below as requested so marking this as done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15)*

**Comment by goldspin** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[ufo/setup_test.py, line 57 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8WRNWmk_liDu7O2UQ_-r1-57:-K8aIZBCL-cO-mhAoc-k:502326265) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/d4aaa57875e4010295f9bc71b62fda84e8ae5d30/ufo/setup_test.py#L57)):*
Sorry, if I wasn't clear.  But I was hoping to see a test where the user isAdmin is False, and we test that the execution flow actually goes to the 'Credentials do not have admin access' error.  Also, this seems to me should be in a separate test method, such as testPostSetupByNonAdminUser or something of that nature.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15)*

**Comment by xwsxethan** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[ufo/setup_test.py, line 57 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8WRNWmk_liDu7O2UQ_-r1-57:-K8aLN8NC8uCZkrBZTsE:-2145448808) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/d4aaa57875e4010295f9bc71b62fda84e8ae5d30/ufo/setup_test.py#L57)):*
If you check out the most recent diff from a couple hours ago, this is already there in the next method down. I even named it testPostSetupNonAdmin haha.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15)*

**Comment by goldspin** [comment]
:lgtm:  Nice!

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

*[ufo/setup_test.py, line 57 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8WRNWmk_liDu7O2UQ_-r1-57:-K8aOJj376-89JaP9m4d:-794584920) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/d4aaa57875e4010295f9bc71b62fda84e8ae5d30/ufo/setup_test.py#L57)):*
Oh, I see it now.  I have to adjust File Matrix myself for the right diff.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-:-K8aOZ7m2xFmhIeJTDc8:-19811245)*

**Comment by xwsxethan** [comment]
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15)*

**Comment by xwsxethan** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[ufo/setup_test.py, line 57 \[r1\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8WRNWmk_liDu7O2UQ_-r1-57:-K8agu7a5h5cUvFrZC8k:-499146009) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/d4aaa57875e4010295f9bc71b62fda84e8ae5d30/ufo/setup_test.py#L57)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15)*

**Comment by jpevarnek** [comment]
Reviewed 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[ufo/setup_test.py, line 26 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ahFOmkafQ3eWOfIGQ:-K8ahFOmkafQ3eWOfIGR:-1290074300) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L26)):*
Do we actually want to do this?  

---

*[ufo/setup_test.py, line 37 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ahOJS3fyM-BJ_3pFx:-K8ahOJS3fyM-BJ_3pFy:-269377290) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L37)):*
My normal objection about checking for specific text that's about to be shown to the user.  Your call if you think this is fine though.

---

*[ufo/setup_test.py, line 45 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8aheimdIkRxYE_HoiY:-K8aheimdIkRxYE_HoiZ:1968585624) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L45)):*
Why does this need to be faked?  It should be sufficient to check that the parameter is a url and then rely upon the google library to give the correct url if called.

---

*[ufo/setup_test.py, line 51 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ahsvRKVn9UpdfKEgA:-K8ahsvRKVn9UpdfKEgB:1127025375) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L51)):*
Honestly not really sure we need to be checking the exact value of config.  As it is, these all seem to just check that the exact value we expect, determined in the exact method we expect, is reflected.

---

*[ufo/setup_test.py, line 57 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8aja2Q9K-2TAdtTmm2:-K8aja2Q9K-2TAdtTmm3:443250213) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L57)):*
This again feels a bit like we're mocking out the whole flow of the implementation method by method, any way to be a bit more generic about this (e.g. an authenticated client will request the user, it will get something that has a different domain, there will be an error)

---

*[ufo/setup_test.py, line 92 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ajuMfjGtqjBab7qKo:-K8ajuMfjGtqjBab7qKp:596170602) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L92)):*
Still not sure why this is mocked out...

---

*[ufo/setup_test.py, line 142 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ak3LuisR1qjgLG3Vi:-K8ak3LuisR1qjgLG3Vj:1884657860) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L142)):*
Let's check that the value got saved instead of just checking that save got called.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15)*

**Comment by xwsxethan** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[ufo/setup_test.py, line 26 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ahFOmkafQ3eWOfIGQ:-K8tqTFktHEvr-0Y1F2I:2046595509) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L26)):*
No, we do not. Good call.

---

*[ufo/setup_test.py, line 37 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ahOJS3fyM-BJ_3pFx:-K8anX467W93XjS_wO_R:-1276930891) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L37)):*
Totally understood and agree. I just wrote it this way to have something, but I fully plan on changing this around once I do the i18n stuff. At least I'm pointing to a constant now instead of hard coded text though.

---

*[ufo/setup_test.py, line 45 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8aheimdIkRxYE_HoiY:-K8arIe2_BKbMjhskwdM:-499146009) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L45)):*
Done.

---

*[ufo/setup_test.py, line 51 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ahsvRKVn9UpdfKEgA:-K8ayPESFI6OLchTAc7E:-340152657) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L51)):*
Okay, I changed it to similarly assert that it is not none like the oauth url. I'll address the exact call portions shortly, but need to do some refactoring first.

---

*[ufo/setup_test.py, line 57 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8aja2Q9K-2TAdtTmm2:-K8tlHMC-DdU0XJ4u8Mk:1866896487) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L57)):*
So the other way to do this since I turned off the config setup is to mock out the call to step2_exchange. I can't submit a valid oauth code with the request without storing it somewhere or doing my own oauth dance in each test (which I'm not even sure I could do). I can submit a fake oauth code, mock out the exchange of the token so that it passes, and continue past that. I don't know what else we can do here without a real user account for the Google Plus API and a real oauth code, but if you know of a way to get the oauth code in the test then I'd be up for that.

---

*[ufo/setup_test.py, line 92 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ajuMfjGtqjBab7qKo:-K8ayMTfqJim2kF48LUj:-499146009) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L92)):*
Done.

---

*[ufo/setup_test.py, line 142 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8ak3LuisR1qjgLG3Vi:-K8tsfVue_LzU6zrAvwI:-499146009) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L142)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15)*

**Comment by xwsxethan** [comment]
@jpevarnek PTAL at comments, as I think this is mostly done.

---

Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-:-K9820SECPDCQf26CYLp:-554401104)*

**Comment by jpevarnek** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[ufo/setup_test.py, line 57 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8aja2Q9K-2TAdtTmm2:-K98QOzHktBH5-LmSBw2:-1078552972) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L57)):*
Go with whatever you think looks best here.  This is probably more a sign of my code being terrible than anything else.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15)*

**Comment by xwsxethan** [comment]
If anyone wants to quickly check this over again and give me another thumbs up before I merge, that would be awesome. Thanks!

---

Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[ufo/setup_test.py, line 57 \[r2\]](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-K8aja2Q9K-2TAdtTmm2:-K9CkyjhGhuL-Dmgn4_G:1005538281) ([raw file](https://github.com/uproxy/ufo-management-server-flask/blob/f5a7fee418616f3ee164e00953ebc93340c27817/ufo/setup_test.py#L57)):*
I'm just going to keep it as is and update as necessary for now. If this changes in the future then I'll be happy to revisit.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/uproxy/ufo-management-server-flask/15#-:-K9ClCj4WQGn9qFUS06n:-539155687)*

**Comment by blueandgold** [comment]
Nice work, looks good to me!  :+1:

**Pull request merged**